### PR TITLE
Deprecate `LinesStream` in favor of `Stream<String>`

### PR DESCRIPTION
**Pull request by basil**
#3211 introduced a `LinesStream` and a corresponding `TextFile#linesStream()` method that produces it. This seems redundant, since Java already has `File#lines` which produces `Stream<String>`. This change deprecates our custom functionality in favor of the standard paradigm. There was only one consumer that had to be migrated.

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
LGTM

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
